Tasks/SystemsManagerGetParameter/TaskOperations.ts

```diff
@@ -73,15 +78,18 @@ export class TaskOperations {
                 })
                 .promise()
 
-            for (const p of response.Parameters) {
-                const outputVariableName = transformParameterToVariableName(this.taskParameters, p.Name)
-                const isSecret = p.Type === 'SecureString'
-                console.log(tl.loc('SettingVariable', outputVariableName, p.Name, isSecret))
+            if (!response.Parameters) {
+                nextToken = undefined
```

Comment:
It looks like you'd want to unconditionally call `nextToken = response.NextToken` and then conditionally do the other stuff below if `response.Parameters` is truthy

Reply:
we can just exit the loop if there are no parameters, so changed it to do that

Reply:
What happens if response.Parameters is not set, but response.NextToken is set?

Reply:
then we break? you are looking a the old version

Reply:
What I meant to ask was "is this a valid situation in which we should use the nextToken to make the next call (ie: continue instead of break)

Reply:
Ah, in that case I don't think so. Parameters can be empty, but since that field is an array that will not cause it to break. 

The message in that case would not be valid. In these kind of cases, we really need better error messages as we have a lot of silent failures like this in the code base.

Reply:
I added an additional error message for this case. I don't feel confident enough that it can't happen to make the task fail, but the error message should at least be better than it silently failing.